packages/@aws-cdk/aws-appconfig-alpha/lib/deployment-strategy.ts

```diff
@@ -0,0 +1,284 @@
+      resource: 'deploymentstrategy',
+      resourceName: deploymentStrategyId,
+    });
+
+    class Import extends Resource implements IDeploymentStrategy {
+      public readonly deploymentStrategyId = deploymentStrategyId;
+      public readonly deploymentStrategyArn = deploymentStrategyArn;
+    }
+
+    return new Import(scope, id, {
+      environmentFromArn: deploymentStrategyArn,
+    });
+  }
+
+  /**
+   * The name of the deployment strategy.
+   */
+  public readonly name?: string;
+
+  /**
+   * The deployment duration in minutes of the deployment strategy.
+   */
+  public readonly deploymentDurationInMinutes?: number;
+
+  /**
+   * The growth factor of the deployment strategy.
+   */
+  public readonly growthFactor?: number;
+
+  /**
+   * The description of the deployment strategy.
+   */
+  public readonly description?: string;
+
+  /**
+   * The final bake time in minutes of the deployment strategy.
+   */
+  public readonly finalBakeTimeInMinutes?: number;
+
+  /**
+   * The growth type of the deployment strategy.
+   */
+  public readonly growthType?: GrowthType;
+
+  /**
+   * The ID of the deployment strategy.
+   */
+  public readonly deploymentStrategyId: string;
+
+  /**
+   * The Amazon Resource Name (ARN) of the deployment strategy.
+   */
+  public readonly deploymentStrategyArn: string;
+
+  private readonly _cfnDeploymentStrategy: CfnDeploymentStrategy;
+
+  constructor(scope: Construct, id: string, props: DeploymentStrategyProps) {
+    super(scope, id, {
+      physicalName: props.name,
+    });
+
+    this.deploymentDurationInMinutes = props.rolloutStrategy.deploymentDuration.toMinutes();
+    this.growthFactor = props.rolloutStrategy.growthFactor;
+    this.description = props.description;
+    this.finalBakeTimeInMinutes = props.rolloutStrategy.finalBakeTime?.toMinutes();
+    this.growthType = props.rolloutStrategy.growthType;
+    this.name = props.name || Names.uniqueResourceName(this, {
+      maxLength: 64,
+      separator: '-',
+    });
+
+    const resource = new CfnDeploymentStrategy(this, 'Resource', {
+      name: this.name,
+      deploymentDurationInMinutes: this.deploymentDurationInMinutes,
+      growthFactor: this.growthFactor,
+      replicateTo: 'NONE',
+      description: this.description,
+      finalBakeTimeInMinutes: this.finalBakeTimeInMinutes,
+      growthType: this.growthType,
+    });
+    this._cfnDeploymentStrategy = resource;
+
+    this.deploymentStrategyId = this._cfnDeploymentStrategy.ref;
+    this.deploymentStrategyArn = this.stack.formatArn({
+      service: 'appconfig',
+      resource: 'deploymentstrategy',
+      resourceName: this.deploymentStrategyId,
+    });
+  }
+}
+
+/**
+ * Defines the growth type of the deployment strategy.
+ */
+export enum GrowthType {
+  LINEAR = 'LINEAR',
+  EXPONENTIAL = 'EXPONENTIAL',
+}
+
+/**
+ * Defines the deployment strategy ID's of AWS AppConfig predefined strategies.
+ */
+export enum PredefinedDeploymentStrategyId {
```

Comment:
I kept this enum here, in case customers wanted to directly import these existing and AppConfig authored deployment strategies, instead of creating new ones through `RolloutStrategy.ALL_AT_ONCE`, etc. Not super necessary but it might add some flexibility for customers who want to directly use the existing authored deployment strategies, rather than creating new ones.

Reply:
I see. Thanks.